### ISSUE TITLE
req.file always undefined

### ISSUE THREAD

**Issue by MHDMAM** [expressjs/multer#210]
Hi,
I'm pretty new to `multer` this my first time using it.

I looked into some resources before asking here but no luck.

here's my code snippet:

```js
var express = require('express');
var multer = require('multer');
//.......
var app = express();
app.use(multer({
	dest: './uploads/'
}).single('file'));
// .....

router.post('/uploadPic', function(req, res) {
	console.log('uploadPic', req.file, req.files, req.body);
});

// Angular code:
//......
		$http.post(ApiUrls.user.uploadPic, {
			file: formData // new FormData(); with image file appended. 
		}, {
			transformRequest: angular.identity,
			headers: {
				'Content-Type': undefined
			}
		});
//.....
```
any idea why `req.file` always undefined.
do I missed anything?

**Comment by LinusU** [comment]
If angular is using jQuery (and the same problem probably exists in their jQLite) they won't handle `FormData` properly. See #193

**Comment by MHDMAM** [comment]
Right, Thanks i'll look into it, at least my `multer` code is correct

**Comment by MHDMAM** [comment]
A'right, I figured it out. I was missing some form' tag attributes.
here's some explanation in case someone else need it.

Well, Angular code up there is correct as well as node part.

``` HTML
<form id="uploadForm" enctype="multipart/form-data" action="/path/to/uploadPic", method="post"
....
</form>
```

Regarding to angular code:
- Angular’s default `transformRequest` function will try to serialize FormData object, so we override it with the identity function to leave the data intact.
- The default Content-Type header for POST and PUT requests is application/json, so we changed by setting ‘Content-Type’: undefined.
